### PR TITLE
[VL-128] Preload endpoint checks

### PR DESCRIPTION
**Pull request by kin0992**
Add the checks related to the PreLoad operation in TC-SEND-01 case.

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/domain/__tests__/PreLoadRepository.test.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import * as data from './data';
+import { contentTypeIsPdf } from '../PreLoadRepository';
+
+describe('PreLoadRepository', () => {
```
Comment (datalek) [inline src/domain/__tests__/PreLoadRepository.test.ts:4]: Add missing test about `uniquePreloadIdx`

File: src/domain/checklist/sendPaymentNotificationChecklist.ts [hunk 2]
```diff
@@ -16,7 +19,13 @@ const group = {
...
   group,
   name: `Exist at least two requests 'Request an "upload slot"' that matches the criteria`,
-  eval: RA.some(() => true),
+  eval: pipe(
+    flow(
+      RA.filterMap(isPreLoadRecord),
+      RA.filter(pipe(existsApiKey, P.and(uniquePreloadIdx), P.and(contentTypeIsPdf))),
+      (records) => RA.size(records) >= 2
+    )
+  ),
```
Comment (datalek) [inline src/domain/checklist/sendPaymentNotificationChecklist.ts:28]:
```suggestion
  eval: flow(
    RA.filterMap(isPreLoadRecord),
    RA.filter(pipe(existsApiKey, P.and(uniquePreloadIdx), P.and(contentTypeIsPdf))),
    (records) => RA.size(records) >= 2
  ),
```

**Review by datalek — CHANGES_REQUESTED** [review]
🦇

File: src/domain/PreLoadRepository.ts [hunk 3]
```diff
@@ -31,3 +38,17 @@ export const makePreLoadRecord = (record: Omit<PreLoadRecord, 'type'>): PreLoadR
 });
 
 export type PreLoadRecordRepository = Repository<PreLoadRecord>;
+
+export const uniquePreloadIdx = (record: PreLoadRecord) =>
```
[the remark this answers is not in the document]
Comment (kin0992) [reply src/domain/PreLoadRepository.ts]: Done!

File: src/domain/PreLoadRepository.ts [hunk 4]
```diff
@@ -31,3 +38,17 @@ export const makePreLoadRecord = (record: Omit<PreLoadRecord, 'type'>): PreLoadR
...
+  pipe(
+    record.input.body,
+    RA.map((elem) => elem.preloadIdx),
+    RA.uniq(s.Eq),
+    (unique) => unique.length === record.input.body.length
+  );
+
+export const contentTypeIsPdf = (record: PreLoadRecord) =>
```
[the remark this answers is not in the document]
Comment (kin0992) [reply src/domain/PreLoadRepository.ts]: Done!

File: src/domain/checklist/sendPaymentNotificationChecklist.ts (hunk 2 above)
Comment (kin0992) [reply src/domain/checklist/sendPaymentNotificationChecklist.ts]: Done!

File: src/domain/__tests__/PreLoadRepository.test.ts (hunk 1 above)
Comment (kin0992) [reply src/domain/__tests__/PreLoadRepository.test.ts:4]: You're right! I'm going to add the tests later!

Comment (kin0992) [reply src/domain/__tests__/PreLoadRepository.test.ts:4]: Done!

File: src/domain/PreLoadRepository.ts [hunk 5]
```diff
@@ -1,9 +1,13 @@
+import * as O from 'fp-ts/Option';
+import { pipe } from 'fp-ts/function';
```
Comment (datalek) [inline src/domain/PreLoadRepository.ts:2]:
```suggestion
import { pipe } from 'fp-ts/function';
import * as O from 'fp-ts/Option';
```

**Review by datalek — APPROVED** [review]
🐤

**Pull request merged**